File: react-native/storyly-placement-react-native/android/src/main/java/com/storylyplacementreactnative/common/SPLogManager.kt
package com.storylyplacementreactnative.common

import com.appsamurai.storyly.core.listener.log.STRLog
import com.appsamurai.storyly.core.listener.log.StorylyLogLevel


object SPLogManager {

    fun setLogLevel(level: String) {
        STRLog.logLevel = decodeLogLevel(level)
    }
}

internal fun decodeLogLevel(level: String): StorylyLogLevel = when (level.lowercase()) {
    "debug" -> StorylyLogLevel.DEBUG
    "warning" -> StorylyLogLevel.WARNING
    "error" -> StorylyLogLevel.ERROR
    else -> StorylyLogLevel.OFF
}

File: flutter/storyly_placement_flutter/ios/Classes/Common/SPLogManager.swift
import Foundation
import StorylyCore

class SPLogManager {

    static func setLogLevel(_ level: String) {
        STRLog.logLevel = decodeLogLevel(level)
    }
}

internal func decodeLogLevel(_ level: String) -> StorylyLogLevel {
    switch level.lowercased() {
    case "debug": return .debug
    case "warning": return .warning
    case "error": return .error
    default: return .off
    }
}
